File: DSM2/TPI/ATV1_HAIMON_CUGLER_VIEIRA/Passagem/src/main/java/com/mycompany/passagem/Passagem.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.passagem;

import java.util.Scanner;

/**
 *
 * @author fatec-dsm2
 */
public class Passagem {

    private String nomePassageiro;
    private String telefone;
    private String rg;
    private String localViagem;
    private String data;
    private String horario;
    private String numeroPoltrona;

    public String getNomePassageiro() {
        return nomePassageiro;
    }

    public void setNomePassageiro(String nomePassageiro) {
        this.nomePassageiro = nomePassageiro;
    }

    public String getTelefone() {
        return telefone;
    }

    public void setTelefone(String telefone) {
        this.telefone = telefone;
    }

    public String getRg() {
        return rg;
    }

    public void setRg(String rg) {
        this.rg = rg;
    }

    public String getLocalViagem() {
        return localViagem;
    }

    public void setLocalViagem(String localViagem) {
        this.localViagem = localViagem;
    }

    public String getData() {
        return data;
    }

    public void setData(String data) {
        this.data = data;
    }

    public String getHorario() {
        return horario;
    }

    public void setHorario(String horario) {
        this.horario = horario;
    }

    public String getNumeroPoltrona() {
        return numeroPoltrona;
    }

    public void setNumeroPoltrona(String numeroPoltrona) {
        this.numeroPoltrona = numeroPoltrona;
    }

    public void cadastrarDadosPassageiro() {

        Scanner scanner = new Scanner(System.in);

        System.out.println("Insira o nome");
        setNomePassageiro(scanner.nextLine());

        System.out.println("Insira o telefone");
        setTelefone(scanner.nextLine());

        System.out.println("Insira o RG");
        setRg(scanner.nextLine());

    }

    public void cadastrarDadosPassagem(String numeroPoltrona) {

        Scanner scanner = new Scanner(System.in);
        
        setNumeroPoltrona(numeroPoltrona);

        System.out.println("Insira o local da viagem");
        setLocalViagem(scanner.nextLine());
        
        System.out.println("Insira a data");
        setData(scanner.nextLine());

        System.out.println("Insira o horario");
        setHorario(scanner.nextLine());

    }
    
    public String mostrarDadosPassageiro(){
        
        return "Nome: " + getNomePassageiro() +
                "\nTelefone: " + getTelefone() +
                "\nRG: " + getRg();
        
    }
    
    public void mostrarDadosPassagem(){
        System.out.println("Local da viagem: " + getLocalViagem() + 
                "\nData: " + getData() + 
                "\nHorario: " + getHorario() + 
                "\nPoltrona: " + getNumeroPoltrona());
    }

}

File: DSM3/TP2/aula04_OO/exercicios/Passagem/Passagem.py
class Passagem:
    def __init__(self):
        self.__nomePassageiro = ""
        self.__telefone = ""
        self.__rg = ""
        self.__localViagem = ""
        self.__data = ""
        self.__hoario = ""
        self.__numpoltrona = ""

    @property
    def _nomePassageiro(self):
        return self.__nomePassageiro

    @_nomePassageiro.setter
    def _nomePassageiro(self, value):
        self.__nomePassageiro = value

    @property
    def _telefone(self):
        return self.__telefone

    @_telefone.setter
    def _telefone(self, value):
        self.__telefone = value

    @property
    def _rg(self):
        return self.__rg

    @_rg.setter
    def _rg(self, value):
        self.__rg = value

    @property
    def _localViagem(self):
        return self.__localViagem

    @_localViagem.setter
    def _localViagem(self, value):
        self.__localViagem = value

    @property
    def _data(self):
        return self.__data

    @_data.setter
    def _data(self, value):
        self.__data = value

    @property
    def _hoario(self):
        return self.__hoario

    @_hoario.setter
    def _hoario(self, value):
        self.__hoario = value

    @property
    def _numpoltrona(self):
        return self.__numpoltrona

    @_numpoltrona.setter
    def _numpoltrona(self, value):
        self.__numpoltrona = value

    def cadastrarDadosPassageiro(self):
        print("\n==Cadastro de passageiro==")
        self.__nomePassageiro = input("Insira o nome: ")
        self.__telefone = input("Insira o telefone: ")
        self.__rg = input("Insira o RG: ")
        

    def cadastrarDadosPassagem(self):
        print("\n==Cadastro da passagem===")
        self.__localViagem = input("Insira o local: ")
        self.__data = input("Insira a data: ")
        self.__hoario = input("Insira o horario: ")
        self.__numpoltrona = input("Insira o numero da polterona: ")

    def mostrarDadosPassageiro(self):
        print("==Dados do Passageiro==")
        print(f"""
        Nome: {self.__nomePassageiro}
        Telefone: {self.__telefone}
        RG: {self.__rg}
        """)

    def mostrarDadosPassagem(self):
        print("==Dados da Passagem==")
        print(f"""
            Local: {self.__localViagem}
            Data: {self.__data}
            Horário: {self.__hoario}
            Poltrona: {self.__numpoltrona}
            """)
